packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -88,6 +89,68 @@ export const onSingleSelectHandler = <T extends Record<string, unknown>>(
   return newState;
 };
 
+/**
+ * Handles selection when clicked on a row while shift key is pressed.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance?: TableInstance<T>,
+  onSelect?: (
+    selectedData: T[] | undefined,
+    tableState?: TableState<T>,
+  ) => void,
+  isRowDisabled?: (rowData: T) => boolean,
+) => {
+  if (instance == null) {
+    return state;
+  }
+
+  let startIndex = instance.flatRows.findIndex(
+    (row) => row.id === state.lastSelectedRow,
+  );
+  let endIndex = instance.flatRows.findIndex((row) => row.id === action.id);
+
+  if (startIndex < 0) {
```

Comment:
Nit: you also can write `let startIndex = Math.max(0, instance.flatRows.findIndex(
    (row) => row.id === state.lastSelectedRow,
  ));`

Reply:
**[DONE]** Thanks